Bose layout. Header and nav blocks created

ZupaFly opened the pull request:
[DEMO LINK](https://zupafly.github.io/layout_miami/)

qoa11a reviewed (changes requested):
The page isn't finished. Please, request review when it's done.

ZupaFly commented:
> The page isn't finished. Please, request review when it's done.

Yes, i am still working on it. Just following "Readme" file requirement:
- when you finish the first block of your page deploy it and create a Pull Request with a [DEMO LINK] etc...

polosanya reviewed (changes requested):
The demo link doesn't work. Please fix it
![image](https://user-images.githubusercontent.com/96409563/216275732-bc79341c-8997-4693-ac53-4e71a2c8eaa0.png)

ZupaFly commented:
> The demo link doesn't work. Please fix it ![image](https://user-images.githubusercontent.com/96409563/216275732-bc79341c-8997-4693-ac53-4e71a2c8eaa0.png)

Done, sorry for that my internet connection is literally bad. Deploying 6-7 times.

polosanya reviewed (changes requested):
Keep up the good work! Check your demo link and make sure that all the links and icons work correctly on click)
1. Add a favicon for your page
![image](https://user-images.githubusercontent.com/96409563/216561771-558bb609-82f1-4c1e-ae35-d3a4a460a4cf.png)
2. There is a 404 error on logo click.
Same with cross in burger menu
![image](https://user-images.githubusercontent.com/96409563/216561866-c83769b0-ac7d-4b1b-883b-f487ac7aa0b3.png)
![image](https://user-images.githubusercontent.com/96409563/216562327-6a53a83e-2a5c-4ffa-ab32-7ba9f1910f78.png)
3. Background color doesn't match the design
![image](https://user-images.githubusercontent.com/96409563/216563206-bf7c7ceb-b10b-4069-969e-2410342bdc9c.png)
![image](https://user-images.githubusercontent.com/96409563/216563406-634157b9-493c-4b33-a317-b9cbc2c51602.png)
4. Same problem with other sections
![image](https://user-images.githubusercontent.com/96409563/216563588-faaf6d8c-382c-4108-af0f-d738d91acdb8.png)
![image](https://user-images.githubusercontent.com/96409563/216563634-8d34ff02-bed6-4d15-87ac-daeb75de9045.png)
5. 404 error on images click. You can make them just images, not links. And add increasing hover effects to all the images on page
![image](https://user-images.githubusercontent.com/96409563/216563907-a4d7f043-0853-4223-a7c1-b7e5250d4431.png)
5. 405 error on form submit
![image](https://user-images.githubusercontent.com/96409563/216564060-a4dc93e9-75ea-4790-93c6-ebf633ace99c.png)
6. Add hover effects to these links and make sure that google maps link opens in a new tab
![image](https://user-images.githubusercontent.com/96409563/216564161-e6e1b57e-6956-443d-b6ac-a0ab72dbb625.png)
7. Implement tablet and mobile versions according to the design

ZupaFly commented:
> Keep up the good work! Check your demo link and make sure that all the links and icons work correctly on click)
> 
> 1. Add a favicon for your page
>    ![image](https://user-images.githubusercontent.com/96409563/216561771-558bb609-82f1-4c1e-ae35-d3a4a460a4cf.png)
> 2. There is a 404 error on logo click.
>    Same with cross in burger menu
>    ![image](https://user-images.githubusercontent.com/96409563/216561866-c83769b0-ac7d-4b1b-883b-f487ac7aa0b3.png)
>    ![image](https://user-images.githubusercontent.com/96409563/216562327-6a53a83e-2a5c-4ffa-ab32-7ba9f1910f78.png)
> 3. Background color doesn't match the design
>    ![image](https://user-images.githubusercontent.com/96409563/216563206-bf7c7ceb-b10b-4069-969e-2410342bdc9c.png)
>    ![image](https://user-images.githubusercontent.com/96409563/216563406-634157b9-493c-4b33-a317-b9cbc2c51602.png)
> 4. Same problem with other sections
>    ![image](https://user-images.githubusercontent.com/96409563/216563588-faaf6d8c-382c-4108-af0f-d738d91acdb8.png)
>    ![image](https://user-images.githubusercontent.com/96409563/216563634-8d34ff02-bed6-4d15-87ac-daeb75de9045.png)
> 5. 404 error on images click. You can make them just images, not links. And add increasing hover effects to all the images on page
>    ![image](https://user-images.githubusercontent.com/96409563/216563907-a4d7f043-0853-4223-a7c1-b7e5250d4431.png)
> 6. 405 error on form submit
>    ![image](https://user-images.githubusercontent.com/96409563/216564060-a4dc93e9-75ea-4790-93c6-ebf633ace99c.png)
> 7. Add hover effects to these links and make sure that google maps link opens in a new tab
>    ![image](https://user-images.githubusercontent.com/96409563/216564161-e6e1b57e-6956-443d-b6ac-a0ab72dbb625.png)
> 8. Implement tablet and mobile versions according to the design

Thanks for your comments. Below improvements 1 by 1:
1. Favicon added. I did't find favicon image from Bose layout design. So, I used Miami layout's favicon.
2. Changed links. Now working correctly. Mistake was, that they were leading to github home page, not my landing page home
3. Done. bg-color changed
4. Done bg-color changed
5. Done all images now have animation. Links improved and leading to correct place.
6. Done. I've changed button type to submit. Now have error 404. As far as i understand that should be ok, because there is no server to processing information.
7. Done. All links with animation on them. And google maps link opening in new tab.
8. Done

polosanya reviewed (changes requested):
1. Form shouldn't be submitted if some of the fields are not filled.
Page shouldn't be reloaded on form submit (https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault)
![image](https://user-images.githubusercontent.com/96409563/216821201-8d2041a0-c53b-41ca-9d46-989c419e475e.png)
2. Make sure that you added transition for the default state of an element, not hover, so hover effects work both ways smoothly
![smooth](https://user-images.githubusercontent.com/96409563/216821487-81434ab8-c078-4822-a1e9-768f5a557e36.gif)
3. 404 error on this link click
![image](https://user-images.githubusercontent.com/96409563/216821512-d8684bde-419d-41e6-b2e2-47fbe2eb483a.png)
5. Mobile version doesn't match the design
![image](https://user-images.githubusercontent.com/96409563/216821707-df35fcf6-abbe-4866-a28c-abb11c507f30.png)
Same with the tablet version. Check yourself using DevTools

ZupaFly commented:
> 1. Form shouldn't be submitted if some of the fields are not filled.
>    Page shouldn't be reloaded on form submit (https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault)
>    ![image](https://user-images.githubusercontent.com/96409563/216821201-8d2041a0-c53b-41ca-9d46-989c419e475e.png)
> 2. Make sure that you added transition for the default state of an element, not hover, so hover effects work both ways smoothly
>    ![smooth](https://user-images.githubusercontent.com/96409563/216821487-81434ab8-c078-4822-a1e9-768f5a557e36.gif)
>          
>            [
>              
>            
>                ![smooth](https://user-images.githubusercontent.com/96409563/216821487-81434ab8-c078-4822-a1e9-768f5a557e36.gif)
>              ](https://user-images.githubusercontent.com/96409563/216821487-81434ab8-c078-4822-a1e9-768f5a557e36.gif)
>            
>            
>              
>                
>                  
>                
>                
>                  
>                  
>                
>              
>              [
>                
>                  
>                
>              ](https://user-images.githubusercontent.com/96409563/216821487-81434ab8-c078-4822-a1e9-768f5a557e36.gif)
> 3. 404 error on this link click
>    ![image](https://user-images.githubusercontent.com/96409563/216821512-d8684bde-419d-41e6-b2e2-47fbe2eb483a.png)
> 4. Mobile version doesn't match the design
>    ![image](https://user-images.githubusercontent.com/96409563/216821707-df35fcf6-abbe-4866-a28c-abb11c507f30.png)
>    Same with the tablet version. Check yourself using DevTools

Thanks for your comments. Below improvements step by step:
1. Done. Now button type is "submit" but in JS i used event.preventDefault, so now it behaves like button type = "button". I am not sure that it was the final requirement, so waiting for your comments about it. Also added _required_ status to all form fields. 
2. Done. Now transition behaves smoothly both ways.
3. Fixed the link.
4. I used figma design as a reference. And there are 3 types of the screen 1260/740/320. All of that resolutions are fitting design pretty well, but sure going between them can looks like a mess. That's why 375px screen respond to 740px and looks bad. I guess for more precise matching i should have more design inputs, or maybe I am doing something wrong. 

Thanks, and waiting for your comments.

Volodymyr-Melnyk-Fron10der reviewed (changes requested):
1. Email should be opened in email client on click event
![image](https://user-images.githubusercontent.com/122989585/219307901-cfb15486-ef78-4e5b-a235-478880591718.png)
2. Form has to be cleaned after submitting + add required attributes to the inputs
![image](https://user-images.githubusercontent.com/122989585/219308288-086901ea-b101-4848-ae8e-183aa06c85fd.png)
3. Implement checkpoint
![image](https://user-images.githubusercontent.com/122989585/219308582-004e5001-ed07-4b8a-b729-2d3dc4ffcf25.png)
4. 404 error after click on phone number in menu
![image](https://user-images.githubusercontent.com/122989585/219308740-e158300d-be8c-47fb-9d69-67d0a633cf24.png)
5. There is horizontal scrolling, get rid of it pls
![image](https://user-images.githubusercontent.com/122989585/219309070-85c74956-ce04-467b-8eb0-6128e68c03bd.png)

ZupaFly commented:
> 4\. 404 error after click on phone number in menu

Thanks for your comments. Below improvements step by step:
1. Emails and phones now opening in their client. Added all the necessary attributes.
2. There are _required_ attributes in all form sections buy I stuck with the problem of submitting from. Either I'm turning off its default behavior with preventDefault event, which is preventing me from using button and I can't see any restrictions if form not filed in or I have all the notifications, but I am refreshing page with 404 error.
3. Done
4. Also have problems with that part. I can't understand what I should do. All the scrolls appearing between design guidelines.  When i am checking default design requirements 1260/740/320 there is no horizontal scroll. But moving between that resolutions making that scroll appear.
Still i do commit changes on point 1 and 3 and i will join Q&A to try to understand my mistakes on point 2 and 4.

Thanks

qoa11a reviewed (changes requested):
![image](https://user-images.githubusercontent.com/102293862/219364298-c0a7329e-058f-4d63-be85-4fe61f8b2203.png)
- here's an example of how to handle form submit
- your page shouldn't have horizontal scroll on any device. Breakpoints in design are for popular variants, but nevertheless, the page shouldn't be scrolled horizontally anyway. You should make your layout adaptive, not working only on 3 exact widths

ZupaFly commented:
> ![image](https://user-images.githubusercontent.com/102293862/219364298-c0a7329e-058f-4d63-be85-4fe61f8b2203.png)
> 
> * here's an example of how to handle form submit
> * your page shouldn't have horizontal scroll on any device. Breakpoints in design are for popular variants, but nevertheless, the page shouldn't be scrolled horizontally anyway. You should make your layout adaptive, not working only on 3 exact widths

Thanks for your comments:
1. Finally figured out how to stop refreshing page on submit. But in my case i used IDs to solve the problem. Now everything works perfectly.
2. I made some changes and updated to design. So now it's smooth even between guidelines. Also, there is no more horizontal scroll line. 
Waiting for comments and have a great day.
